src/cunumeric/stat/histogram_omp.cc

```diff
@@ -0,0 +1,209 @@
+/* Copyright 2023 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/stat/histogram.h"
+#include "cunumeric/stat/histogram_template.inl"
+
+#define _USE_THRUST_
+
+#ifdef _USE_THRUST_
+#include "cunumeric/stat/histogram_cpu.h"
+#include "cunumeric/stat/histogram_impl.h"
+#endif
+
+#include <omp.h>
+
+#include <algorithm>
+#include <numeric>
+#include <tuple>
+
+namespace cunumeric {
+using namespace legate;
+
+namespace detail {
+
+// RO accessor (size, pointer) extractor:
```

Comment:
Notice that the 2 `get_accessor_ptr()` are the same for all branches, not `make_accessor_copy()`. So, I could factor out `get_accessor_ptr()` into `histogram_gen.h`, as I'd like to keep `histogram_impl.h` clean, only with histogram implementation details, and oblivious to accessors and anything not related to the implementation. Agree?

Reply:
`histogram_gen.h` is a fine place to move them, I just want to reduce duplication where it makes sense.

Note that `make_accessor_copy()` can also be made uniform. The only difference is that on the GPU version you explicitly specify `Legion::Memory::Kind::GPU_FB_MEM` to `create_buffer`, which isn't necessary because that's the default when running on a GPU thread.

Reply:
Done according to the 4 points we agreed:

1. unify accessor helpers to one undifferentiated form for every ecosystem (GPU/CPU/OMP);
2. put them in a header (*_gen.h), under a deeper nested namespace, say, cunumeric::detail::accessor namespace;
3. move the copy inside make_accessor_copy();
4. do the copying via thrust::copy_n();